fe/fe-core/src/main/java/com/starrocks/sql/optimizer/RequiredPropertyDeriver.java

```diff
@@ -145,9 +145,15 @@ public Void visitPhysicalMergeJoin(PhysicalMergeJoinOperator node, ExpressionCon
 
     @Override
     public Void visitPhysicalNestLoopJoin(PhysicalNestLoopJoinOperator node, ExpressionContext context) {
-        PhysicalPropertySet rightBroadcastProperty =
-                new PhysicalPropertySet(new DistributionProperty(DistributionSpec.createReplicatedDistributionSpec()));
-        requiredProperties.add(Lists.newArrayList(PhysicalPropertySet.EMPTY, rightBroadcastProperty));
+        if (node.getJoinType().isRightJoin() || node.getJoinType().isFullOuterJoin()) {
+            PhysicalPropertySet rightGather =
```

Comment:
add comment to descibe why right join need gather not broadcast in nest-loop?

Reply:
done